infrastructure/src/main/java/org/corfudb/infrastructure/AbstractServer.java

```diff
@@ -36,10 +39,66 @@ public AbstractServer() {
      * @param r   The router that took in the message.
      */
     public void handleMessage(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
+        boolean isEnabled = MetricsUtils.isMetricsCollectionEnabled();
+        Timer.Context context = null;
+
         if (isShutdown()) return;
+        switch (msg.getMsgType()) {
```

Comment:
I am not happy with this long/manual switch. Once again, service components should be able to register themselves to the metering infrastructure. This is more modular, and scopes everything relevant to a component within that file. Advantages include, for example, that there is no need to change in multiple places when components change. 

Similarly to the way different message types are handled by the different servers, AbstractServer should have a generic function like getHandler(). It can return an object similar to CorfuMsgHandler, which has methods  'getTimeContext(msg-type..)' . And so on.